Constant crashes in Release build (current beta in Google Play Store)

When generating a signed APK with the ProdRelease build, I get these errors:

```
C:\Code\GitHub\apps-android-commons\app\build.gradle
Error:Error: commons-logging defines classes that conflict with classes now provided by Android. Solutions include finding newer versions or alternative libraries that don't have the same problem (for example, for httpclient use HttpUrlConnection or okhttp instead), or repackaging the library using something like jarjar. [DuplicatePlatformClasses]
Error:Error: httpclient defines classes that conflict with classes now provided by Android. Solutions include finding newer versions or alternative libraries that don't have the same problem (for example, for httpclient use HttpUrlConnection or okhttp instead), or repackaging the library using something like jarjar. [DuplicatePlatformClasses]
```

I was able to carry on and generate the APK regardless, but we should still probably look into them so that we don't keep getting errors in future versions.

Update: Currently, it seems that in order to release any of the new features post-2.5.0, we will need to:

- [ ] Fix issue with commons-logging and httpclient
- [ ] Fix error with kaptProdReleaseKotlin
- [ ] Re-test ProdRelease build (I am hoping we can find a way for other devs to test release builds as well, see #988

Ran logcat:

```
11-27 22:31:35.631 25616-25616/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: fr.free.nrw.commons, PID: 25616
                                                   java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.contributions.ContributionsActivity}: java.lang.NullPointerException: fr.free.nrw.commons.CommonsApplication.activityInjector() returned null
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2924)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2985)
                                                       at android.app.ActivityThread.-wrap14(ActivityThread.java)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1635)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:154)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6692)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
                                                    Caused by: java.lang.NullPointerException: fr.free.nrw.commons.CommonsApplication.activityInjector() returned null
                                                       at dagger.internal.Preconditions.checkNotNull(Preconditions.java:79)
                                                       at dagger.android.AndroidInjection.inject(AndroidInjection.java:56)
                                                       at fr.free.nrw.commons.contributions.ContributionsActivity.onCreate(ContributionsActivity.java:142)
                                                       at android.app.Activity.performCreate(Activity.java:6912)
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1126)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2877)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2985) 
                                                       at android.app.ActivityThread.-wrap14(ActivityThread.java) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1635) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:154) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:6692) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358) 

11-27 22:31:40.509 25646-25646/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: fr.free.nrw.commons, PID: 25646
                                                   java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.contributions.ContributionsActivity}: java.lang.NullPointerException: fr.free.nrw.commons.CommonsApplication.activityInjector() returned null
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2924)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2985)
                                                       at android.app.ActivityThread.-wrap14(ActivityThread.java)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1635)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:154)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6692)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
                                                    Caused by: java.lang.NullPointerException: fr.free.nrw.commons.CommonsApplication.activityInjector() returned null
                                                       at dagger.internal.Preconditions.checkNotNull(Preconditions.java:79)
                                                       at dagger.android.AndroidInjection.inject(AndroidInjection.java:56)
                                                       at fr.free.nrw.commons.contributions.ContributionsActivity.onCreate(ContributionsActivity.java:142)
                                                       at android.app.Activity.performCreate(Activity.java:6912)
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1126)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2877)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2985) 
                                                       at android.app.ActivityThread.-wrap14(ActivityThread.java) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1635) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:154) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:6692) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358) 
11-27 22:31:41.343 3698-4045/? E/OpenGLRenderer: Error:glDeleteShader::<shader> is not a value generated by OpenGL
11-27 22:31:41.343 3698-4045/? E/OpenGLRenderer: Error:glDeleteShader::<shader> is not a value generated by OpenGL
11-27 22:31:41.343 3698-4045/? E/OpenGLRenderer: Error:glDeleteShader::<shader> is not a value generated by OpenGL
11-27 22:31:41.343 3698-4045/? E/OpenGLRenderer: Error:glDeleteShader::<shader> is not a value generated by OpenGL
11-27 22:31:41.343 3698-4045/? E/OpenGLRenderer: Error:glDeleteShader::<shader> is not a value generated by OpenGL
11-27 22:31:41.343 3698-4045/? E/OpenGLRenderer: Error:glDeleteShader::<shader> is not a value generated by OpenGL
11-27 22:31:41.356 3698-3786/? E/ViewRootImpl: sendUserActionEvent() mView == null
11-27 22:31:41.390 25666-25666/? E/Zygote: v2
11-27 22:31:41.391 25666-25666/? E/Zygote: accessInfo : 0
11-27 22:31:41.943 8461-8461/? E/SKBD KeyboardInfoUtils: getInstance start
11-27 22:31:41.944 8461-8461/? E/SKBD KeyboardInfoUtils: sendSIPInformation state:6  isAbstractKeyboardView : true
11-27 22:31:41.946 8461-25714/? E/SKBD KeyboardInfoUtils: sending null keyboardInfo as SIP is closed
11-27 22:31:41.982 8461-8461/? E/SamsungIME: <AbstractKeyboardView> - onDraw() called
11-27 22:31:42.629 8461-8461/? E/SKBD KeyboardInfoUtils: getInstance start
11-27 22:31:42.629 8461-8461/? E/SKBD KeyboardInfoUtils: sendSIPInformation state:6  isAbstractKeyboardView : true
11-27 22:31:42.630 8461-25715/? E/SKBD KeyboardInfoUtils: sending null keyboardInfo as SIP is closed
11-27 22:31:42.800 8461-25717/? E/SamsungIME: updateShiftState() : 2
11-27 22:31:42.907 8461-8461/? E/SamsungIME: <AbstractKeyboardView> - onDraw() called
11-27 22:31:42.997 8461-8461/? E/SamsungIME: <AbstractKeyboardView> - onDraw() called
11-27 22:31:44.239 3698-3794/? E/BatteryStatsService: no controller energy info supplied
11-27 22:31:44.257 3698-3794/? E/BatteryStatsService: modem info is invalid: ModemActivityInfo{ mTimestamp=0 mSleepTimeMs=0 mIdleTimeMs=0 mTxTimeMs[]=[0, 0, 0, 0, 0] mRxTimeMs=0 mEnergyUsed=0}

```

The source of crash is definitely dagger but am not sure why it is happening just on release builds. I tried disabling prodguard and minification on release build(just for finding the root cause) but that too didnt help. 

This is the stacktrace. 

```
FATAL EXCEPTION: main
Process: fr.free.nrw.commons, PID: 5632
java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.contributions.ContributionsActivity}: java.lang.NullPointerException: fr.free.nrw.commons.CommonsApplication.activityInjector() returned null
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3160)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3275)
at android.app.ActivityThread.access$1000(ActivityThread.java:218)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1744)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:145)
at android.app.ActivityThread.main(ActivityThread.java:7007)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
Caused by: java.lang.NullPointerException: fr.free.nrw.commons.CommonsApplication.activityInjector() returned null
at dagger.internal.Preconditions.checkNotNull(Preconditions.java:79)
at dagger.android.AndroidInjection.inject(AndroidInjection.java:56)
at fr.free.nrw.commons.contributions.ContributionsActivity.onCreate(ContributionsActivity.java:142)
at android.app.Activity.performCreate(Activity.java:6609)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1134)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3113)
```

@whym @psh Do you guys have any insights on it?

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -31,13 +31,6 @@
                   android:excludeFromRecents="true"
                   android:finishOnTaskLaunch="true" />
 
-        <activity
-            android:name="org.acra.CrashReportDialog"
-            android:excludeFromRecents="true"
-            android:finishOnTaskLaunch="true"
-            android:launchMode="singleInstance"
-            android:theme="@android:style/Theme.Dialog" />
-
         <activity android:name=".auth.LoginActivity">
             <intent-filter>
                 <category android:name="android.intent.category.LAUNCHER" />

--- a/app/src/main/java/fr/free/nrw/commons/CommonsApplication.java
+++ b/app/src/main/java/fr/free/nrw/commons/CommonsApplication.java
@@ -67,6 +67,7 @@ public class CommonsApplication extends DaggerApplication {
     public void onCreate() {
         super.onCreate();
 
+        Fresco.initialize(this);
         if (setupLeakCanary() == RefWatcher.DISABLED) {
             return;
         }
@@ -81,8 +82,6 @@ public class CommonsApplication extends DaggerApplication {
 
         // Fire progress callbacks for every 3% of uploaded content
         System.setProperty("in.yuvi.http.fluent.PROGRESS_TRIGGER_THRESHOLD", "3.0");
-
-        Fresco.initialize(this);
     }
 
     protected RefWatcher setupLeakCanary() {

